tao_compiler/mlir/disc/transforms/lower_tf.cc

```diff
@@ -1342,6 +1342,65 @@ class ConvertSparseSegmentMeanOp
   }
 };
 
+class ConvertWhereOp : public OpRewritePattern<TF::WhereOp> {
+ public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TF::WhereOp op,
+                                PatternRewriter& rewriter) const override {
+    auto loc = op.getLoc();
+    auto where = rewriter.create<mhlo_disc::WhereOp>(
+        loc, op.index().getType(),
+        RankedTensorType::get({1}, rewriter.getI64Type()), op.input());
+
+    // num_output_elements
+    Value num_output_elements_vec = where.getResult(1);
+    Value idx_zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value idx_one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+    Value num_output_elements = rewriter.create<arith::IndexCastOp>(
+        loc, rewriter.getIndexType(),
+        rewriter.create<tensor::ExtractOp>(loc, num_output_elements_vec,
+                                           idx_zero));
+    Value input_dims =
+        rewriter.create<tensor::DimOp>(loc, where.getResult(0), 1);
+
+    // output is rank 2
+    SmallVector<Value, 2> start_values(2, idx_zero);
+    SmallVector<Value, 2> limit_values(2, input_dims);
+    limit_values[0] = num_output_elements;
+    SmallVector<Value, 2> strides_values(2, idx_one);
```

Comment:
Nit: I think the following codes are more human readable.
```
ArrayRef<Value> start_values{idx_zeros, idx_zeros};
ArrayRef<Value> limit_values{num_output_elements, input_dims};
ArrayRef<Value> strides_values{idx_one, idx_one):
```

Reply:
It's not safe use ArrayRef here?

Reply:
I think they can be passed to the parameters directory, since they will not be modified.

Reply:
ArrayRef does not owned any container space. The temporary initialized list is destroyed after constructing the array ref.